engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/inputSettings/InputSettingsScreen.java

```diff
@@ -94,6 +86,35 @@ public void initialise() {
         mainLayout.setVerticalSpacing(8);
         mainLayout.setFamily("option-grid");
 
+        UIButton azerty = new UIButton();
```

Comment:
This would probably work better using a `UIDropdown` instead as that's designed to choose between multiple options, which is what's being done here.

Reply:
I agree a `UIDropdown` would feel more natural. I think it should be designed with a different behavior than the button and the console command.

At the moment, "Apply" only changes a set of mappings. Other mappings are untouched.
That means if I blindly replaced the buttons by a dropdown, moving from one state to another and then back would not give the same result.
For example, moving from AZERTY to NEO then back to AZERTY would leave some NEO mappings that were applied. In my opinion that is not what a user would expect from a dropdown.

For that reason I think I will change the behavior to restore defaults all keys before applying the special keyboard bindings when selecting a new keyboard state.